src/flag_gems/ops/cummin.py

```diff
@@ -0,0 +1,341 @@
+
+from ..utils import libentry
+
+
+@triton.jit
+def _min_combine(a, idxa, b, idxb):
+    if a < b:
+        return a, idxa
+    else:
+        return b, idxb
+
+
+@triton.jit
+def tl_cummin(input, index, axis=0):
+    return core.associative_scan((input, index), axis, _min_combine)
+
+
+@triton.jit
+def _argmin_combine_tie_break_right(value1, index1, value2, index2):
+    tie = value1 == value2 and index1 > index2
+    lt = value1 < value2 or tie
+    value_ret = core.where(lt, value1, value2)
+    index_ret = core.where(lt, index1, index2)
+    return value_ret, index_ret
+
+
+@triton.jit
+def tl_min_tie_break_right(input, axis=None, keep_dims=False):
+    return core._reduce_with_indices(
+        input, axis, _argmin_combine_tie_break_right, keep_dims=keep_dims
+    )
+
+
+@libentry()
+@triton.jit(do_not_specialize=["n_elements"])
+def add_base_min_kernel(
+    out,
+    out_indices,
+    partial_min,
+    partial_min_indices,
+    n_elements,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    offset = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = offset < n_elements
+
+    out_ptrs = out + offset
+    out_indices_ptrs = out_indices + offset
+    out_vals = tl.load(out_ptrs, mask=mask)
+    out_indices = tl.load(out_indices_ptrs, mask=mask)
+
+    if pid > 0:
+        partial_min_ptrs = partial_min + pid - 1
+        last_part_min_via_min = tl.load(partial_min_ptrs)
+        partial_min_indices_ptrs = partial_min_indices + pid - 1
+        last_part_min_index_via_min = tl.load(partial_min_indices_ptrs)
+
+        final_vals = tl.minimum(out_vals, last_part_min_via_min)
+        final_indices = tl.where(
+            out_vals <= last_part_min_via_min, out_indices, last_part_min_index_via_min
+        )
+        tl.store(out_ptrs, final_vals.to(out_vals.dtype), mask=mask)
+        tl.store(out_indices_ptrs, final_indices, mask=mask)
+
+
+@libentry()
+@triton.jit(do_not_specialize=["n_elements"])
+def scan_part_min_kernel(
+    inp,
+    out,
+    out_indices,
+    partial_min,
+    partial_min_indices,
+    n_elements,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    offset = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = offset < n_elements
+
+    inp_ptrs = inp + offset
+    inp_vals = tl.load(inp_ptrs, mask=mask)
+    if (
+        tl.constexpr(inp_vals.dtype.is_int64())
+        or tl.constexpr(inp_vals.dtype.is_uint64())
+    ) or tl.constexpr(inp_vals.dtype.is_fp64()):
+        inp_vals = inp_vals
+    elif tl.constexpr(inp_vals.dtype.is_int()):
+        inp_vals = inp_vals.to(tl.int32)
+    else:
+        inp_vals = inp_vals.to(tl.float32)
+    result, cummin_indices = tl_cummin(inp_vals, offset, axis=0)
+
+    # tl.min do not support min_indices_tie_break_right
+    part_min_via_min, part_min_indices_via_min = tl_min_tie_break_right(
+        inp_vals, axis=0
+    )
+    # part_min_via_min = result[BLOCK_SIZE - 1]
+    # part_min_indices_via_min = indices[BLOCK_SIZE - 1]
```

Comment:
Why I can not index from `result`, I want to get `part_min_via_min` directly from `result` rather than get min from `inp_vals`
![image](https://github.com/user-attachments/assets/7a759c8a-5105-48b3-9304-2238d9728ba6)

Reply:
Triton does not support indexing.